expression/builtin_time.go

```diff
@@ -3577,6 +3674,29 @@ func (b *builtinAddDateStringDecimalSig) evalTime(row chunk.Row) (types.Time, bo
 	return result, isNull || err != nil, err
 }
 
+func (b *builtinAddDateStringDecimalSig) evalString(row chunk.Row) (string, bool, error) {
```

Comment:
Why not wrap this function as:
return addDateString(row, type)

func addDateString ..... {
   XXX
  switch type {
 set interval
 }
 XXX
}